src/components/text/text.tsx

```diff
@@ -0,0 +1,61 @@
+import { ReactNode } from "react";
+import cc from "classcat";
+import "./text.scss";
+
+export type TextVariant = "link" | "title" | "paragraph" | "small" | "default";
+export type TextColor =
+  | "primary"
+  | "secondary"
+  | "tertiary"
+  | "black"
+  | "white";
+
+export interface IProps {
+  variant?: TextVariant;
+  className?: string;
+  children: string | ReactNode;
+  link?: string;
+  headingLevel?: keyof JSX.IntrinsicElements;
+  color?: TextColor;
+}
+
+export const Text = ({
+  variant = "default",
+  className,
+  children,
+  link,
+  headingLevel = "h2",
+  color = "black",
+}: IProps) => {
+  const commonProps = {
+    className: cc(["text", variant, headingLevel, className, color]),
+    children,
+  };
+
+  switch (variant) {
```

Comment:
I'm not sure that a separate component is needed for it. 

Probably, it would be better to create a set of mixins. 

If you want to go with a separate component, we need to have the ability to pass element type via props. 
For example, in the interface we will have something like `as?: ElementType<HTMLAttributes<HTMLElement>>;`, and in the component we will render it like `as: Tag = "div"; <Tag {...delegatedProps}/ >`. Usage will be `<Text as="span"/>`